src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -224,18 +228,24 @@ public void AddTagHelperAttribute(string name, object value)
         }
 
         /// <summary>
-        /// Execute and retrieve the rendered child content asynchronously.
+        /// Execute children asynchronously with the given <paramref name="encoder"/> in scope and return their
+        /// rendered content.
         /// </summary>
+        /// <param name="useCachedResult">If <c>true</c> multiple calls to this method will not cause re-execution
+        /// of child content; cached content will be returned.</param>
+        /// <param name="encoder">
+        /// The <see cref="HtmlEncoder"/> to use. Does not override all HTML encodings which may occur.
+        /// </param>
         /// <returns>A <see cref="Task"/> that on completion returns the rendered child content.</returns>
         /// <remarks>
         /// Child content is only executed once. Successive calls to this method or successive executions of the
         /// returned <see cref="Task{TagHelperContent}"/> return a cached result.
         /// </remarks>
-        public async Task<TagHelperContent> GetChildContentAsync(bool useCachedResult)
+        public async Task<TagHelperContent> GetChildContentAsync(bool useCachedResult, HtmlEncoder encoder)
```

Comment:
Hmm, i thought we were going to use `null` to indicate we shouldn't override anything on the page. With this approach we can special case `null` in the `HtmlEncoder` `GetChildContentAsync` override to pass in a non-encoding encoder.

The other alternative would be to pass through additional parameters indicating if you truly want encoding. Neither way is that nice.